admin/model/extension/module/altapay.php

```diff
@@ -53,5 +63,44 @@ public function updateOrderMeta($order_id, $capture = false, $refund = false, $v
             $this->db->query("UPDATE " . DB_PREFIX . "altapay_orders SET modified='".$this->db->escape((string)date('Y-m-d H:i:s', time()))."', $status_query WHERE order_id='".(int)$order_id."'");
         }
 	}
-	
+
+    /**
+     * @param int $order_id
+     *
+     * @return array
+     */
+   public function getOrderReconciliationIdentifiers($order_id)
+    {
+        $query = $this->db->query('SELECT reconciliation_identifier, transaction_type FROM `' . DB_PREFIX . 'altapay_order_reconciliation` WHERE order_id =' . (int) $order_id);
```

Comment:
Casting a string will give you a `0` as a result, causing side effects for unvalidated order ids

Reply:
I've updated such that the SQL will only execute if `order_id` passes integer validation > 0.